modules/installation-osp-creating-control-plane.adoc

```diff
@@ -0,0 +1,139 @@
+    with_indexed_items: "{{ [os_port_master] * os_cp_nodes_number }}"
+    register: ports
+
+  - name: 'Set Control Plane ports tag'
+    command:
+      cmd: "openstack port set --tag {{ cluster_id_tag }} {{ item.1 }}-{{ item.0 }}"
+    with_indexed_items: "{{ [os_port_master] * os_cp_nodes_number }}"
+
+  - name: 'List the Control Plane Trunks'
+    command:
+      cmd: "openstack network trunk list"
+    when: os_networking_type == "Kuryr"
+    register: control_plane_trunks
+
+  - name: 'Create the Control Plane trunks'
+    command:
+      cmd: "openstack network trunk create --parent-port {{ item.1.id }} {{ os_cp_trunk_name }}-{{ item.0 }}"
+    with_indexed_items: "{{ ports.results }}"
+    when:
+    - os_networking_type == "Kuryr"
+    - "os_cp_trunk_name|string not in control_plane_trunks.stdout"
+
+  - name: 'Get a token for creating the server group'
+    os_auth:
+    register: cloud
+
+  # If microversion 2.15 is not available, replace with:
+  # - name: 'Create the Control Plane server group'
+  #   uri:
+  #     method: POST
+  #     headers:
+  #       X-Auth-Token: "{{ cloud.ansible_facts.auth_token }}"
+  #       X-OpenStack-Nova-API-Version: '2.64'
+  #     url: "{{ cloud.ansible_facts.service_catalog | selectattr('name', 'match', 'nova') | first | json_query('endpoints') | selectattr('interface', 'match', 'public') | first | json_query('url') }}/os-server-groups"
+  #     body_format: json
+  #     body:
+  #       server_group:
+  #         name: "{{ os_cp_server_group_name }}"
+  #         policy: "soft-anti-affinity"
+  #   register: cp_group
+  - name: 'Create the Control Plane server group'
+    uri:
+      method: POST
+      headers:
+        X-Auth-Token: "{{ cloud.ansible_facts.auth_token }}"
+        X-OpenStack-Nova-API-Version: '2.15'
+      url: "{{ cloud.ansible_facts.service_catalog | selectattr('name', 'match', 'nova') | first | json_query('endpoints') | selectattr('interface', 'match', 'public') | first | json_query('url') }}/os-server-groups"
+      body_format: json
+      body:
+        server_group:
+          name: "{{ os_cp_server_group_name }}"
+          policies:
+          - soft-anti-affinity
+    register: cp_group
+
+  - name: 'Create the Control Plane servers'
+    os_server:
+      name: "{{ item.1 }}-{{ item.0 }}"
+      image: "{{ os_image_rhcos }}"
+      flavor: "{{ os_flavor_master }}"
+      auto_ip: no
+      # The ignition filename will be concatenated with the Control Plane node
+      # name and its 0-indexed serial number.
+      # In this case, the first node will look for this filename:
+      #    "{{ infraID }}-master-0-ignition.json"
+      userdata: "{{ lookup('file', [item.1, item.0, 'ignition.json'] | join('-')) | string }}"
+      nics:
+      - port-name: "{{ os_port_master }}-{{ item.0 }}"
+      scheduler_hints:
+        group: "{{ cp_group.json.server_group.id }}"
+    with_indexed_items: "{{ [os_cp_server_name] * os_cp_nodes_number }}"
```

Comment:
Server groups are a 4.5 feature. Do you want to add them anyway?

FYI this is the 4.4 version of the file, we are actively backporting every change that impacts 4.4 UPI:
https://github.com/openshift/installer/blob/release-4.4/upi/openstack/04_control-plane.yaml

Reply:
Good catch! I jumped the gun. I'm going to revert https://github.com/openshift/openshift-docs/commit/030502c27185512972b3e6e32a1c6e14416ccc77 for now and check that everything is in line with release-4.4.